src/user-management/interfaces/authentication-event.interface.ts

```diff
@@ -0,0 +1,32 @@
+interface AuthenticationEventError {
+  code: string;
+  message: string;
+}
+
+export type AuthenticationEventType =
+  | 'email_verification'
+  | 'magic_auth'
+  | 'mfa'
+  | 'oauth'
+  | 'password'
+  | 'sso';
+
+export interface AuthenticationEvent<T extends AuthenticationEventType> {
```

Comment:
Note for tomorrow: I think we should export separate interfaces per event type, this will allow us to expand for the case where individual ones have additional fields in the future. 

```ts
type AuthenticationPasswordEvent = AuthenticationEvent<'password'>
```

Could be expanded to:
```ts
type AuthenticationPasswordEvent = AuthenticationEvent<'password'> & { additionalField: 'true' }
```

Reply:
i think for type-specific fields we can do something like what stripe does for their `PaymentMethod`: https://github.com/stripe/stripe-node/blob/7293b69a204c4a25ca00574078f6b6699936b3cf/types/PaymentMethods.d.ts#L25

Reply:
Hmmm didn't understand, are they using namespaces to override/extend fields from the base type?

Reply:
here's the API reference: https://docs.stripe.com/api/payment_methods/object

`PaymentMethod` objects have a `type` field with values like `card` or `affirm`. any data specific to the payment method type is then stored in a field named after the `type` value. so a `PaymentMethod` with a `type` of the value `card` will have a `card` field containing card-specific data.

the way this is implemented in the SDK is a `PaymentMethod` interface with optional fields corresponding to each of the `type` values. in the link above, `card` and `affirm` are optional fields on `PaymentMethod`s.

when we eventually have data specific to the authentication event `type`, we can use a similar pattern -- `sso`, `password`, etc can all be optional on the `AuthenticationEvent`.

the downside of the above approach: this doesn't allow the type system to guarantee that `sso`-type events will have a `sso` field. but i think that's ok for the use cases of these events (observability).

the benefit of the above approach is we have more flexibility to make non-breaking changes, which is useful especially because the failed authentication events are still a work in progress. we can always tighten the types later (a non-breaking change).

Reply:
>  to make non-breaking changes

How it would be a non-breaking change - could you give an example? Sorry, it's not clear to me also how the current interfaces could lead to breaking changes, just trying to understand before moving forward.

Reply:
Introduced a similar approach on https://github.com/workos/workos-node/pull/998/commits/39fdd375ae77605facb75fe92876b675fc9fc351

The Stripe auto-generated types include all event types as properties even as nullable. Here we're just indexing the one equivalent still as nullable to better handle breaking changes:

![CleanShot 2024-03-26 at 14 52 22](https://github.com/workos/workos-node/assets/48022589/063242f1-ed6f-4b43-813d-486f407042b6)

Reply:
Also, it seems that we're diverting a bit from the API response - it's not wrong, but just wanted to point out that this behavior will change in the future if we decide to go for auto-generated APIs, where the API spec will be almost a 1:1 map to the SDK return types. 

This is a symptom that will happen for not only authentication events but all of the other modules.

Reply:
oh sorry, we can skip these type-specific fields for now. the API doesn't return them currently. if/when we do return them we can handle this then.